SwissTransferFeatures/SettingsView/EditSettingView.swift

```diff
@@ -0,0 +1,208 @@
+/*
+ Infomaniak SwissTransfer - iOS App
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import InfomaniakDI
+import STCore
+import STResources
+import SwiftUI
+import SwissTransferCore
+import SwissTransferCoreUI
+
+struct EditCellDatom: Hashable, Equatable {
+    let label: String
+    let action: () -> Void
+    let index: Int
+    let leftIconAsset: STResourcesImages?
+
+    init(label: String, action: @escaping () -> Void, index: Int, leftIconAsset: STResourcesImages? = nil) {
+        self.label = label
+        self.action = action
+        self.index = index
+        self.leftIconAsset = leftIconAsset
+    }
+
+    static func == (lhs: Self, rhs: Self) -> Bool {
+        lhs.label == rhs.label
+    }
+
+    func hash(into hasher: inout Hasher) {
+        hasher.combine(label)
+    }
+}
+
+// TODO: Refactor
+struct EditSettingViewDatasource {
+    @LazyInjectService private var settingsManager: AppSettingsManager
+
+    let source: SettingDetailUi
+
+    var title: String {
+        switch source {
+        case .theme:
+            return STResourcesStrings.Localizable.settingsThemeTitle
+        case .validityPeriod:
+            return STResourcesStrings.Localizable.settingsValidityPeriodTitle
+        case .downloadLimit:
+            return STResourcesStrings.Localizable.settingsDownloadsLimitTitle
+        case .emailLanguage:
+            return STResourcesStrings.Localizable.settingsEmailLanguageTitle
+        default:
+            return ""
+        }
+    }
+
+    func validityPeriod(index: Int) -> ValidityPeriod {
```

Comment:
I'll refactor the index stuff. Probably will use some ad-hoc polymorphism instead. Hope it will not be seen as "too complicated".